### ISSUE TITLE
An error occured while converting texture ->  Visual C++ Runtime 2008 RTM

### ISSUE THREAD

**Issue by Kunstbanause** [RareExports/Banjo-s-Backpack#3]
### 1) Visual C++ Runtime 2008 RTM
https://www.microsoft.com/en-us/download/confirmation.aspx?id=26368

### 2) 32 Bit (x86 instead of x64)
I needed to run in x86 explicitly (was set to "any CPU")
![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/4bddc32f-7171-4adb-9f33-6edfefa48705)

![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/a718f7e2-22a5-401b-8c39-92b16892a6d0)

### 3) Missing Resources
Also I needed to copy the _out_, _resources_ and _tmp_ folder from a [release build ](https://github.com/RareExports/Banjo-s-Backpack/releases) into my debug build:
![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/63508f0d-2061-485d-9145-b630d3d6d62d)

## Original Issue
Hey ya! Appreciate the github upload. I have an error when exporting a level and importing it again. Also this error occurs when I import my own levels I made in blender. I hope you can help:

`An error occured while converting texture <path>\image_0012.png, please check all files are present and try again.`
The seemingly missing _image_0012_ is of course where it needs to be, along side the other textures.

The level does import but there are a lot of missing textures:
![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/4a32444a-269a-4e2a-a4ee-b6758295f95a)

I am on Windows 10
The error happens with your version here from git hub: Banjo's Backpack release 1.0.5
as well as with versions from the internet.

**Comment by ParchmentWolf** [comment]
@Zeldaboy14 if you find the error code, ill go on video call with you to figure out the problem.

My assumption is its got a path code setup properly but the get current directory code is jank since microsoft decided the best route is 5+ ways of getting current directory that arent very clear and have subtle ways they can fail based on arbitrary reasons per meens of getting cd.

**Comment by Kunstbanause** [comment]
In ObjImporter.cs line 705 the error is thrown. I don’t have a running debug version as of now, but I‘ll try to dig deeper. 

[ChatGPT](https://chat.openai.com/share/a105f2de-4a29-4f61-8619-0c4986ea16e0)  suggested some improvements to the error handling: 
* Added an additional try-catch block outside the loop to handle any exceptions occurring during the entire texture conversion process.
* Added an inner try-catch block to handle exceptions specific to individual texture file conversions. This way, if an error occurs while processing one texture, it won't stop the conversion process for other textures.
* The exception messages are displayed in the error message boxes, including the specific file path and the exception message.
Improved file path handling by using Path.
* Combine to combine directory and file names.

(Code for that in the gpt link in the last two responses)

**Comment by Kunstbanause** [comment]
> but I‘ll try to dig deeper

Ok I have a running debug build. I found the detailed message to be (translated from german):
`Message "The file or assembly \"Magick.NET-x86, Version=7.0.0.0, Culture=neutral, PublicKeyToken=2004825badfa91ec\" or a dependency on it was not found. An attempt was made to load a file with an incorrect format." string`

**Comment by ParchmentWolf** [comment]
you could look for a nuget package on the magic.net and see if u can pull a package (itd pull necessary dependencies and ensure dependency issues on the library are an invalid concern)

**Comment by Kunstbanause** [comment]
Hmm, I just noticed the `Magick.NET-x86.dll` is in the \bin\Debug folder. So it seems like this might not be the actual error after all...

**Comment by Kunstbanause** [comment]
Currently following the error message to a system expectio: _Expection Settings_ -> _System.BadImageFormatException_

```
if (uniqueColors.Count > 16)
                image = ImageHandler.ConvertTo16Colour(image);
```

So the texture has more than 16 colors and the code tries to convert it down, but then a BadImageFormatException is thrown. But I checked the path, the fileformat and that the image (a .img) is not corrupted which it is not. So I dont know whats up with that

**Comment by Kunstbanause** [comment]
> could look for a nuget package on the magic.net and see if u can pull a package 

I failed to instal them with the cli or download a package but was able to install through the visual studio interface. Unfortunately to no effect in regards to the error.

![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/fb960775-0cae-4afe-b215-82eba29ab5f3)

The version 7.0.0.0 is the one from the error message (I also tried the latest version but that is not compatible with the low .net Framework 3.5)

**Comment by Kunstbanause** [comment]
I tried x64 vs x86 stuff to no effect yet: https://stackoverflow.com/a/62263479/21890329

**Comment by Kunstbanause** [comment]
I got it working, sheeeesh...
![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/f979b841-79eb-4055-8760-28daf3cfe6ea)
Of course a darn ms somthing was missing:
**Visual C++ Runtime 2008 RTM:**
# https://www.microsoft.com/en-us/download/confirmation.aspx?id=26368

**Comment by Kunstbanause** [comment]
@z64me @Zeldaboy14 Not sure, what the right way would be, to include this for everyone, but maybe write it as an dependency into the readme for starters?

**Comment by Kunstbanause** [comment]
I think the vertex colors are missing. But here is an export of an original level and the reimport:
## Original in the Model Viewer
![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/6c4bca8f-583b-4a33-9fc4-da17b1a62f65)

## Reimport in the Model Importer
![grafik](https://github.com/RareExports/Banjo-s-Backpack/assets/3209654/c45a08c9-0b8c-479e-b2d2-a59b0ff502b1)

**Comment by Zeldaboy14** [comment]
the exporter exports to the RGBA obj format. The importer is the old spec GEOBJ spec. Meaning follow the instructions to import to blender and fix accordingly. I want RGBA spec exportation from blender and importation in BB, but need spicy's help to solve it.
Tried to make this as easy as can be, because GEOBJ sucks when importing. If we got RGBA implanted for importation, it'd solve the alpha issues we run into right on the dot
